helm/dagger/templates/engine-statefulset.yaml

```diff
@@ -80,12 +90,28 @@ spec:
               name: {{ include "dagger.fullname" . }}-magicache-token
               {{- end }}
           {{- end }}
+          {{- if .Values.engine.port }}
+          ports:
+            - name: dagger
+              containerPort: {{ .Values.engine.port }}
+              protocol: TCP
+              {{- if .Values.engine.hostPort }}
+              hostPort: {{ .Values.engine.hostPort }}
+              {{- end }}
+          {{- end }}
           securityContext:
             privileged: true
             capabilities:
               add:
                 - ALL
           resources: {{- toYaml .Values.engine.resources | nindent 12 }}
+          {{- if .Values.engine.lifecycle }}
+          lifecycle:
+            {{- if .Values.engine.lifecycle.preStop }}
+            preStop:
+              {{- toYaml .Values.engine.lifecycle.preStop | nindent 14 }}
+            {{- end }}
+          {{- end }}
```

Comment:
Nice addition! I had a use-case a while back where I was using `dagger core engine local-cache prune` to manually free up the disk space on shutdown. FTR: https://docs.dagger.io/configuration/cache/